Shopify/ruby-lsp
# Adjust monorepo configurations

### Motivation

Complete explanation in https://github.com/Shopify/ruby-lsp/pull/1407.

### Implementation

My approach for this PR has been applying the minimal changes necessary so that we can work in the new monorepo structure.

There are many things we can do after this is merged, like

- Improving code workspace configurations
- Consolidating more of our documentation
- Migrating our integration tests that use a fake client to use the actual VS Code extension

My proposal is that we get a minimally working setup shipped and then iterate on it.

### Manual tests

1. Open the `lsp.code-workspace` workspace in VS Code
2. Verify you can see both workspaces (gem and VS Code extension) at the same level
3. Verify you can use launch tasks, such as starting the VS Code extension in development mode (don't forget to yarn install!)
4. Basically, do a brief verification of basic development and try to find anything that seems to be breaking the experience. If nothing major stands out, we can probably ship and iterate on improving it
